qualtran/bloqs/arithmetic/addition.py

```diff
@@ -74,18 +75,26 @@ class Add(Bloq):
         [Halving the cost of quantum addition](https://arxiv.org/abs/1709.06648)
     """
 
-    dtype: Union[QInt, QUInt, QMontgomeryUInt] = field()
+    a_dtype: Union[QInt, QUInt, QMontgomeryUInt] = field()
+    b_dtype: Union[QInt, QUInt, QMontgomeryUInt] = field()
```

Comment:
actually keep the ` = field()` and make this  parameter assumes the value of the `a_dtype` when not provided
```py3
    b_dtype: Union[QInt, QUInt, QMontgomeryUInt] = field()
   
    @b_dtype.default
     def _b_dtype_default(self):
           return self.a_dtype
```

This will  make your change backward compatible and greatly reduce the size of your PR (you will probably be able to undo all of the changes to the old _test.py and elsewhere)

Reply:
Done